packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -184,6 +184,60 @@ class VisualStudio {
     }
   }
 
+  /// The path to cl.exe, or null if no Visual Studio installation has
+  /// the components necessary to build.
+  String? get clPath {
+    return _getMsvcBinPath('cl.exe');
+  }
+
+  /// The path to lib.exe, or null if no Visual Studio installation has
+  /// the components necessary to build.
+  String? get libPath {
+    return _getMsvcBinPath('lib.exe');
+  }
+
+  /// The path to link.exe, or null if no Visual Studio installation has
+  /// the components necessary to build.
+  String? get linkPath {
+    return _getMsvcBinPath('link.exe');
+  }
+
+  String? _getMsvcBinPath(String executable) {
+    final VswhereDetails? details = _bestVisualStudioDetails;
+    if (details == null || !details.isUsable || details.installationPath == null || details.msvcVersion == null) {
+      return null;
+    }
+
+    return _fileSystem.path.joinAll(<String>[
+      details.installationPath!,
+      'VC',
+      'Tools',
+      'MSVC',
+      details.msvcVersion!,
+      'bin',
+      'Hostx64',
+      'x64',
+      executable,
+    ]);
+  }
+
+  /// The path to vcvars64.exe, or null if no Visual Studio installation has
+  /// the components necessary to build.
+  String? get vcvarsPath {
+    final VswhereDetails? details = _bestVisualStudioDetails;
+    if (details == null || !details.isUsable || details.installationPath == null) {
+      return null;
+    }
+
+    return _fileSystem.path.joinAll(<String>[
+      details.installationPath!,
+      'VC',
+      'Auxiliary',
+      'Build',
+      'vcvars64.bat',
```

Comment:
This exists on my Windows Arm64 machine and targets x64 using the x64 host toolchain. @pbo-linaro, we will need to update this to support targeting Arm64.